Tests/SwiftParserTest/Declarations.swift

```diff
@@ -212,6 +212,96 @@ final class DeclarationTests: XCTestCase {
     )
   }
 
+  func testAccessLevelModifier() {
+    AssertParse(
+      """
+      private(set) var a = 0
+      """
+    )
+
+    AssertParse(
+      """
+      private(#^DIAG^#get) var a = 0
+      """,
+      diagnostics: [
+        DiagnosticSpec(message: "Expected 'set' in modifier"),
+        DiagnosticSpec(message: "Unexpected text 'get' in modifier")
+      ]
+    )
+
+    AssertParse(
+      """
+      private(#^DIAG^#+
+        set
+      ) var a = 0
+      """,
+      diagnostics: [
+        DiagnosticSpec(message: "Expected 'set' in modifier"),
+        DiagnosticSpec(message: "Expected ')' to end modifier"),
+        // FIXME: It should print `+` as detail of text.
+        DiagnosticSpec(message: "Unexpected text in variable")
```

Comment:
Just FYI (not any request to change this in this PR): The unexpected text spans multiple lines here:
```
+
  set
```

and that’s why we print the generic “Unexpected text” instead of “Unexpected '+ set'” (which would look ugly if there’s a line break in the diagnostic message).

Reply:
I see.
I think that escaped representation is best in such case.

For example,

```
Unexpected '+\n  set'
```

Reply:
I got an idea.
In this case, can parser split code into `+` and `\n  set` ?
Currently `\n  ` is trailing trivia of `+`.
If they are moved to leading trivia of `set`,
unexpected text is only `+` and `\n  set` is valid code.

Reply:
If you’ve got ideas, I would be very open to discuss them in follow-up PRs. One thing to keep in mind though, is that these lines might be >100 characters long and the unexpected text might span more than 2 lines.

Also, I don’t think we should be showing `\n` in the diagnostic message because it’s non-trivial to read and confusing to newcomers who don’t know about the `\n` escape sequence.